cpp/CMakeLists.txt

```diff
@@ -840,3 +841,79 @@ add_custom_command(OUTPUT CUDF_DOXYGEN
                    VERBATIM)
 
 add_custom_target(docs_cudf DEPENDS CUDF_DOXYGEN)
+
+
+####################################################################################################
+# - cuDatasource - OFF by default due to increased number of dependencies
+if(BUILD_CUDF_KAFKA)
+
+    # cudf_kafka library
+    add_library(libcudf_kafka
+        libcudf_kafka/kafka/src/kafka_consumer.cpp
+    )
+
+    # Include paths
+    include_directories("${CMAKE_SOURCE_DIR}/libcudf_kafka/kafka/include")
+    
+    # Rename installation to proper name for later finding
+    set_target_properties(libcudf_kafka PROPERTIES OUTPUT_NAME "cudf_kafka")
+    set_target_properties(libcudf_kafka PROPERTIES BUILD_RPATH "\$ORIGIN")
+
+    ###################################################################################################
+    # cudf_kafka - librdkafka -------------------------------------------------------------------------
+
+    find_path(RDKAFKA_INCLUDE "librdkafka" HINTS "$ENV{RDKAFKA_ROOT}/include")
+    find_library(RDKAFKA++_LIBRARY "rdkafka++" HINTS "$ENV{RDKAFKA_ROOT}/lib" "$ENV{RDKAFKA_ROOT}/build")
+
+    message(STATUS "RDKAFKA: RDKAFKA++_LIBRARY set to ${RDKAFKA++_LIBRARY}")
+    message(STATUS "RDKAFKA: RDKAFKA_INCLUDE set to ${RDKAFKA_INCLUDE}")
+
+    target_link_libraries(libcudf_kafka ${RDKAFKA++_LIBRARY})
+
+    ###################################################################################################
+    # - cudf_kafka Install ----------------------------------------------------------------------------
+    target_link_libraries(cudf libcudf_kafka)
+
+    install(TARGETS libcudf_kafka
+            DESTINATION lib
+            COMPONENT libcudf_kafka)
+
+    install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/libcudf_kafka/kafka/include
```

Comment:
Given this is just a `libcudf_kafka` package now, should we just make this `libcudf_kafka/include`?

Reply:
If I understand correctly you are recommending just removing the `kafka` subdirectory of `libcudf_kafka` right? Just want to validate before making that change since that would be a couple of other places in the cmake scripts to change.

Reply:
Yea, we're not trying to make a mega library of different IO sources here, we just want a kafka specific library. If we want additional data sources in the future they'll be additional libraries.